code/game/machinery/dna_infuser/dna_infusion.dm

```diff
@@ -0,0 +1,75 @@
+
+///returns a boolean whether a machine occupant can be infused
+/atom/movable/proc/can_infuse(mob/feedback_target)
+	if(feedback_target)
+		balloon_alert(feedback_target, "no dna!")
+	return FALSE
+
+/mob/living/can_infuse(mob/feedback_target)
+	if(feedback_target)
+		balloon_alert(feedback_target, "dna too simple!")
+	return FALSE
+
+/mob/living/carbon/human/can_infuse(mob/feedback_target)
+	// Checked by can_mutate but explicit feedback for this issue is good
+	if(HAS_TRAIT(src, TRAIT_BADDNA))
+		if(feedback_target)
+			balloon_alert(feedback_target, "dna is corrupted!")
+		return FALSE
+	if(!can_mutate())
+		if(feedback_target)
+			balloon_alert(feedback_target, "dna is missing!")
+		return FALSE
+	return TRUE
+
+///returns /datum/infuser_entry that matches an item being used for infusion, returns a fly mutation on failure
+/atom/movable/proc/get_infusion_entry() as /datum/infuser_entry
+	var/datum/infuser_entry/found
+	for(var/datum/infuser_entry/entry as anything in flatten_list(GLOB.infuser_entries))
+		if(entry.tier == DNA_MUTANT_UNOBTAINABLE)
+			continue
+		if(is_type_in_list(src, entry.input_obj_or_mob))
+			found = entry
+			break
+	if(!found)
+		found = GLOB.infuser_entries[/datum/infuser_entry/fly]
+	return found
+
+/// Attempt to replace/add-to the occupant's organs with "mutated" equivalents.
+/// Returns TRUE on success, FALSE on failure.
+/// Requires the target mob to have an existing organic organ to "mutate".
+// TODO: In the future, this should have more logic:
+// - Replace non-mutant organs before mutant ones.
+/mob/living/carbon/human/proc/infuse_organ(datum/infuser_entry/entry)
```

Comment:
Do we want to commit to this being called `infuse organ`?
I know the machine is the infuser, but if we're genericising the proc maybe there's better names out there (mutate organ?)
Willing to accept that this is the best name if you think it is though (it does take an infuser entry datum after all)

Reply:
I'd be fine changing it but stuff like "mutate" is an already commonly used term for all sorts of stuff especially regarding genetics as a whole, infusion is at least relatively unused.

Reply:
And by the way I do agree, just that it also matches infusion entries and the general system has infusion all over it. we can change all that! But it should be a namespace that doesn't get confused with other things

Reply:
Well it seems that neither you nor I know what that namespace is (tbh its definitely not mutate) so I guess we'll merge it for now and we can change it if someone comes up with something better later

Reply:
👍